onnxruntime/core/providers/cuda/tensor/sequence_op.cc

```diff
@@ -63,6 +65,7 @@ ONNX_OPERATOR_KERNEL_EX(
     11,
     kCudaExecutionProvider,
     (*KernelDefBuilder::Create())
+        .InputMemoryType(OrtMemTypeCPUInput, 1)
         .TypeConstraint("S", DataTypeImpl::AllFixedSizeSequenceTensorTypes())
```

Comment:
Side question: what happens in memcpy planning if this optional input is not present? Is it smart enough to handle that? I assume so as there would be nothing to copy across devices.

Reply:
Yeah, the `Exists()` call will return false for that corresponding NodeArg and hence it stops right there